### ISSUE TITLE
The Style/SpaceAroundBlockParameters does not work as documented ?

### ISSUE THREAD

**Issue by olbat** [bbatsov/rubocop#2485]
I'm not sure I understand the documentation of the `Style/SpaceAroundBlockParameters` cop:
```yaml
Style/SpaceAroundBlockParameters:
  Description: 'Checks the spacing inside and after block parameters pipes.'
```
There is also this configuration parameter:
```yaml
Style/SpaceAroundBlockParameters:
  EnforcedStyleInsidePipes: no_space
  SupportedStyles:
    - space
    - no_space
```

So if I understand it properly, if this cop is enabled with the `EnforcedStyleInsidePipes` set to *no_space* it should forbid spaces inside block parameters, but it does not seems to be the case.

Here is an example:
```bash
$ cat rubocop.yml
Style/SpaceAroundBlockParameters:
  Description: 'Checks the spacing inside and after block parameters pipes.'
  Enabled: true
  EnforcedStyleInsidePipes: no_space

$ echo '{}.inject({}){|r, (k, v)| r[v] = k; r}' | rubocop --config .rubocop.yml --only=Style/SpaceAroundBlockParameters --stdin -
Inspecting 1 file
.

1 file inspected, no offenses detected
```
Since there is a space between each parameters of the block (`r, (k, v)`), shouldn't the cop give a warning ?

Thank you in advance.

**Comment by alexdowad** [comment]
> Since there is a space between each parameters of the block (r, (k, v)), shouldn't the cop give a warning ?

No. If it was `{| r, (k, v) | r[v] = k; r}`, then it would warn about that. This cop checks for spaces between the `|` symbol and the first parameter name, and between the last parameter name and the closing `|`.

This issue can be closed.

**Comment by olbat** [comment]
Okay, thanks for your answer :)

BTW, will it be possible to clarify the documentation about that ?
I don't find that `Checks the spacing inside and after block parameters pipes.` is particularely clear.

**Comment by alexdowad** [comment]
> will it be possible to clarify the documentation about that ?

If you can come up with something clearer, it is very possible. What do you suggest?
